test/test_util.py

```diff
@@ -531,8 +531,6 @@ def test_rewind_body_bad_position(self) -> None:
         # Pass non-integer position
         with pytest.raises(ValueError):
             rewind_body(body, body_pos=None)
-        with pytest.raises(ValueError):
-            rewind_body(body, body_pos=object())
```

Comment:
I think we want to keep this test, just change what's passed to `rewind_body`. It would be `_FAILEDTELL` in this case.

Reply:
`rewind_body` does not support the `object()` type anymore, so the test will pass but `mypy` will cry so removed this.

Reply:
It never supported `object()` but the purpose of this test was to ensure a generic "sentinel-like" object would cause a value error rather than being interpreted as `_FAILEDTELL`. Perhaps it's a little less relevant now, but probably still be validated.

It also points to another interesting issue. The typing of `rewind_body` is wrong and appears to have been adjusted to appease these tests. `body_pos` should not be Optional on `rewind_body`, so we may need to fix that as well.

Reply:
What do you suggest here then @nateprewitt ?

Reply:
If we're running `mypy` over the test suite, I'll defer to @SethMLarson, @pquentin on how they want to handle it. It does seem like there should be a way to support testing unsupported types without creating incorrect type signatures.

My proposal would be to determine how we can support this with mypy, update the signature on `rewind_body` to remove the `Optional` definition, and leave this test untouched.

Reply:
> If we're running `mypy` over the test suite, I'll defer to @sethmlarson, @pquentin on how they want to handle it. It does seem like there should be a way to support testing unsupported types without creating incorrect type signatures.

Okey


Removing `Optional` won't solve the `mypy` error as `object()` is not supported in `Union[int, _TYPE_FAILEDTELL]`.

Going ahead with that will impact this test case instead.

```py

with pytest.raises(ValueError):
    rewind_body(body, body_pos=None)
```

Reply:
Correct, I was noting the `None` test case above _appears_ to be what introduced Optional to `rewind_body` in the first place.

Both test cases should be supported without having to mutilate the typing. I'll need to do some more reading on what our options are for this.

Reply:
It may be as simple as using `# type: ignore` in these two test calls. I'll defer to the other contributors on if that's acceptable as I haven't been involved in previous typing discussions.

Reply:
Definitely keep the test cases and use `# type: ignore[arg-type]` to make Mypy happy (or whatever Mypy says is the error code).